platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -17,68 +19,366 @@ components within the platform.
+The term `AddressBook` has been ambiguously used to refer to both the Hedera Address Book and the Platform Address
+Book. The Platform Address Book is a proper subset of the Hedera Address Book that is only concerned with the
+consensus nodes, their consensus weight, gossip endpoints, and related cryptography. The Hedera Address Book
+contains additional information that is not relevant to the platform portion of the consensus nodes, such as the
+account numbers of the nodes, their staked hbar, the proxy endpoints, and the existence of other types of nodes in
+the network. We've decided to disambiguate the two by referring to the Platform Address Book as the `Roster` and
+making their lifecycle distinct from each other. The life-cycle for updating the Roster must also change as a
+pre-requisite to the implementation of Threshold Signature Scheme (TSS) and the generation of network Ledger Ids.
+
+Prior to this proposal, after the completion of the Dynamic Address Book (DAB) Phase 2 effort, the platform
+`AddressBook` is updated by writing out a new config.txt file and updated cryptography to disk in response to a
+freeze upgrade transaction, just before shut down. Out of band, the new config.txt and cryptography are moved to
+replace the previous ones on disk. On software startup, the new config.txt and cryptography are read in and adopted
+during a software upgrade. Discrepancy in the information present between the `Roster` data structure and the
+`AddressBook` data structure make config.txt a poor medium for carrying out `Roster` updates. Writing the updated
+roster out to disk is less secure than storing it in the state. When TSS is implemented, there will be TSS related
+key material stored in the state, and it is imperative that this key material stay in sync with the `Roster` data.
+
+After this proposal is implemented, new rosters will be created and stored in the state for adoption at a software
+upgrade boundary. It has been common practice in testing to use manually created config.txt files to start new
+networks from genesis and transplant merkle state to a different new network from an existing network. To continue to
+allow these processes, alternate mechanisms of providing a genesis roster and a network overriding roster will be
+introduced.
+
+### Dependencies, Interactions, and Implications
+
+The implementation of the TSS-Roster proposal must come after the completion of DAB Phase 2 and must be completed
+before TSS-Ledger-ID.
+
+This change in the mechanics of how the roster is updated, new networks are started, and how merkle state is
+transplanted will impact current DevOps practices significantly. SOLO and all testing environments will need to be
+updated as well. This change includes the inversion of control of the startup process where in the responsibility
+for reading the state, address book, and configuration from disk is moved from the platform to `ServicesMain`.
```

Comment:
Not sure if this is the right place, but it would be nice to explicitly call out that the implementation of much of this feature lies within the bounds of services code as things are currently defined.

Reply:
Added comment here.